pkg/datamover/profile.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datamover
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/kopia"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/kopia/snapshot"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+// Check that Profile implements DataMover interface
+var _ DataMover = (*Profile)(nil)
+
+type Profile struct {
```

Comment:
Do we need to export Profile struct here? We can keep it local I think to avoid confusion with `Profile` keyword being used at other places as well, unless it is being used outside.

Reply:
I think we could get issue at https://github.com/kanisterio/kanister/pull/1951/files#diff-5d3e3ea6759eb194ae4b353ee2d80b406ab30f8f8dac0cc170bc5ad86aa41407R107
where we're returning `Profile` datamover. Since the function is Exported and return type is not exported.
WDYT @ankitjain235 ?

Reply:
Done !